File: Java/dependency-injection/_06_conforming-container-antipattern/src/main/java/com/github/grzesiek_galezowski/ConformingContainerAntipattern/Inbound/BinaryUdpInbound.java
package com.github.grzesiek_galezowski.ConformingContainerAntipattern.Inbound;

import com.github.grzesiek_galezowski.ConformingContainerAntipattern.ApplicationRoot;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Core.ProcessingWorkflow;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Interfaces.InboundMessage;

public class BinaryUdpInbound implements Inbound {
  private final InputSocket socket;
  private final PacketParsing parsing;
  private ProcessingWorkflow processingWorkflow;

  public BinaryUdpInbound() {
    socket = ApplicationRoot.CONTEXT.resolve(InputSocket.class);
    parsing = ApplicationRoot.CONTEXT.resolve(PacketParsing.class);
  }

  public void setDomainLogic(ProcessingWorkflow processingWorkflow) {
    this.processingWorkflow = processingWorkflow;
  }

  public void startListening() {
    byte[] frameData = new byte[100];
    while (socket.receive(frameData)) {
      InboundMessage message = parsing.resultFor(frameData);
      if (message != null) {
        if (processingWorkflow != null) {
          processingWorkflow.applyTo(message);
        }
      }
    }
  }
}

File: CSharp/DependencyInjection/DependencyInjection/ConformingContainerAntipattern/_3_ConformingContainer/Inbound/BinaryUdpInbound.cs
using ConformingContainerAntipattern._3_ConformingContainer.Core;

namespace ConformingContainerAntipattern._3_ConformingContainer.Inbound
{
  internal interface IInbound
  {
    void SetDomainLogic(IProcessingWorkflow processingWorkflow);
    void StartListening();
  }

  class BinaryUdpInbound : IInbound
  {
    private IProcessingWorkflow _processingWorkflow;
    private readonly IInputSocket _socket;
    private readonly IPacketParsing _parsing;

    public BinaryUdpInbound()
    {
      _socket = ApplicationRoot.Context.Resolve<IInputSocket>();
      _parsing = ApplicationRoot.Context.Resolve<IPacketParsing>();
    }

    public void SetDomainLogic(IProcessingWorkflow processingWorkflow)
    {
      _processingWorkflow = processingWorkflow;
    }
    
    public void StartListening()
    {
      byte[] frameData;
      while (_socket.Receive(out frameData))
      {
        var message = _parsing.ResultFor(frameData);
        if (message != null)
        {
          if (_processingWorkflow != null)
          {
            _processingWorkflow.ApplyTo(message);
          }
        }
      }
    }
  }
}